### PR TITLE
 [ClangImporter] Import NSNotificationName constants as `nonisolated`

### PR DESCRIPTION
**Pull request by xedin**
These constants could be used with observer APIs from a different isolation
context, so it's more convenient to import them as `nonisolated` unless
they are explicitly isolated to a MainActor.

Resolves: rdar://114052705

### CODE REVIEW

**Comment by xedin** [comment]
@swift-ci please test

**Comment by xedin** [comment]
@swift-ci please test

File: lib/ClangImporter/ImportDecl.cpp [hunk 1]
```diff
@@ -8138,6 +8138,23 @@ ClangImporter::Implementation::importSwiftAttrAttributes(Decl *MappedDecl) {
...
+  //
+  // These constants could be used with observer APIs from a different isolation
+  // context, so it's more convenient to import them as `nonisolated` unless
+  // they are explicitly isolated to a MainActor.
+  if (!seenMainActorAttr) {
+    auto *DC = MappedDecl->getDeclContext();
+    if (DC->isTypeContext() && isa<VarDecl>(MappedDecl)) {
+      auto *mappedVar = cast<VarDecl>(MappedDecl);
+      if (mappedVar->isStatic() && mappedVar->isLet() &&
+          isNSNotificationName(cast<clang::ValueDecl>(ClangDecl)->getType())) {
```
Comment (beccadax) [inline lib/ClangImporter/ImportDecl.cpp:8152]: This logic should probably be driven off `importer::isNSNotificationGlobal()` or some version of that. There are *many* constants in the SDK that are declared as type `NSString *`, but are imported as `NSNotification.Name` based on their names having a `Notification` suffix—this logic ought to apply to them, too.

**Review by beccadax — CHANGES_REQUESTED** [review]

File: lib/ClangImporter/ImportDecl.cpp (hunk 1 above)
Comment (beccadax) [reply lib/ClangImporter/ImportDecl.cpp:8152]: Sorry, misunderstood the requirements here. We're trying to fix something that only affects notification names that have been `NS_SWIFT_NAME()`d into an actor-isolated type, and `isNSNotificationGlobal()` rejects anything with an `NS_SWIFT_NAME()`.

**Review by beccadax — APPROVED** [review]
After rethinking what this PR is actually trying to accomplish, I think this is *probably* okay.

**Comment by xedin** [comment]
@swift-ci please test

**Pull request merged**